packages/web/src/components/AppFlows/index.jsx

```diff
@@ -1,35 +1,35 @@
 import PropTypes from 'prop-types';
-import { useQuery } from '@apollo/client';
 import { Link, useSearchParams } from 'react-router-dom';
-import { GET_FLOWS } from 'graphql/queries/get-flows';
 import Pagination from '@mui/material/Pagination';
 import PaginationItem from '@mui/material/PaginationItem';
+
 import * as URLS from 'config/urls';
 import AppFlowRow from 'components/FlowRow';
 import NoResultFound from 'components/NoResultFound';
 import useFormatMessage from 'hooks/useFormatMessage';
-const FLOW_PER_PAGE = 10;
-const getLimitAndOffset = (page) => ({
-  limit: FLOW_PER_PAGE,
-  offset: (page - 1) * FLOW_PER_PAGE,
-});
+import useConnectionFlows from 'hooks/useConnectionFlows';
+import useAppFlows from 'hooks/useAppFlows';
+
 function AppFlows(props) {
   const { appKey } = props;
   const formatMessage = useFormatMessage();
   const [searchParams, setSearchParams] = useSearchParams();
   const connectionId = searchParams.get('connectionId') || undefined;
   const page = parseInt(searchParams.get('page') || '', 10) || 1;
-  const { data } = useQuery(GET_FLOWS, {
-    variables: {
-      appKey,
-      connectionId,
-      ...getLimitAndOffset(page),
-    },
-  });
-  const getFlows = data?.getFlows || {};
-  const { pageInfo, edges } = getFlows;
-  const flows = edges?.map(({ node }) => node);
+  const isConnectionFlowEnabled = !!connectionId;
+  const isAppFlowEnabled = !!appKey && !connectionId;
+
+  const connectionFlows = useConnectionFlows(
+    { connectionId, page },
+    { enabled: isConnectionFlowEnabled },
+  );
+
+  const appFlows = useAppFlows({ appKey, page }, { enabled: isAppFlowEnabled });
+
+  const flows = connectionFlows?.data?.data || appFlows?.data?.data || [];
```

Comment:
Let's depend on `isConnectionFlowEnabled` instead of data's existence. The same for the `pageInfo`.